### DISCUSSION TITLE
RTMP / SRT output

### DISCUSSION THREAD

**Discussion by nums** [OvenMediaLabs/OvenMediaEngine#1408] [Ideas]
Hello,

I thought this had already been discussed but I couldn't find the feed.
It would be interesting to be able to retrieve stream in RTMP or SRT.
The main use case is RTMP/SRT => ovenmedia => RTMP/SRT => processing (like opencv etc) => RTMP/SRT => ovenmedia => WebRTC / LLHLS

For now I'm using FFMPEG in listener mode, so it's not blocking, but if it's not complicated it would be nice :)

**Comment by hernanrz** [comment]
You can work around this using the RTMP Push API https://airensoft.gitbook.io/ovenmediaengine/push-publishing#rtmppush-publisher until a rtmp output is added

**Comment by bchah** [comment]
+1 for this important functionality - if an SRT caller could receive a stream rather than just sending it, this would unlock very powerful remote playback capability!

**Comment by bchah** [comment]
Checking in on SRT Listener Publisher. Without it, we are using fragile integrations to other software to achieve SRT listener output, and we aren't really achieving sub-second latency as a result. It would be a huge relief for OME to support it since it is more and more a customer requirement for sub-second latency *outside of the browser*. Any updates on this?

**Reply by getroot** [reply]
Now that OME has been updated to support MPEG-2 TS containers, we just need to implement the transport part. But the biggest problem is that I am too busy. I am aiming to implement it within this year.

**Reply by naanlizard** [reply]
Rtmp output will be very useful for generating screenshots. 

Is there any word on improved performance for the screenshot/thumbnail provider?
